packages/contentful--create-contentful-app/src/template.ts

```diff
@@ -0,0 +1,82 @@
+import { resolve } from 'path';
+import { existsSync, readFileSync } from 'fs';
+import degit from 'degit';
+import rimraf from 'rimraf';
+
+import { CLIOptions, ContentfulTemplate } from './types';
+import { highlight, warn } from './logger';
+import { rmIfExists } from './utils';
+
+function isContentfulTemplate(url: string) {
+  return Object.values(ContentfulTemplate).some(t => url.includes('contentful/apps/templates/' + t));
+}
+
+function makeContentfulTemplateSource(options: CLIOptions) {
+  const name = options.Js ? ContentfulTemplate.Javascript : ContentfulTemplate.Typescript;
+  return `contentful/apps/templates/${name}`;
+}
+
+function getTemplateSource(options: CLIOptions) {
+  const source = options.templateSource ?? makeContentfulTemplateSource(options);
+
+  if (!isContentfulTemplate(source)) {
+    warn(`Template at ${highlight(source)} is not an official Contentful app template!`);
+  }
+
+  return source;
+}
+
+async function clone(source: string, destination: string) {
+  const d = degit(source, { mode: 'tar' });
+
+  try {
+    await d.clone(destination);
+  } catch (e: any) {
+    let message = e.message ?? 'Unknown error';
+    if (e.code === 'DEST_NOT_EMPTY') {
+      message = 'Destination directory is not empty.';
+    }
+    throw new Error(message);
+  }
+}
+
+function validate(destination: string): void {
+  const packageJSONLocation = `${destination}/package.json`;
```

Comment:
Did I get it right — if we'll try to clone a non-existent folder, there will be no package.json?

Reply:
Exactly

Reply:
![image](https://user-images.githubusercontent.com/8289447/154033770-72284c4c-fc41-4bf5-8ffd-13d068d5d41a.png)